doc/nginx-cert-gke.md

```diff
@@ -0,0 +1,387 @@
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    # Enable the HTTP-01 challenge provider
+    solvers:
+      - http01:
+          ingress:
+            class: nginx
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-v02.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: user@example.com
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-prod
+    # Enable the HTTP-01 challenge provider
+    solvers:
+      - http01:
+          ingress:
+            class: nginx
+```
+
+Once created, apply the custom resource:
+```bash
+kubectl apply -f issuers.yaml
+```
+
+## Step 6) Deploy XMTPD with an Ingress
+
+These helm charts provide convenient templates for cert-manager and nginx.
+
+Fetch the repo:
+```bash
+git clone git@github.com:xmtp/xmtpd-infrastructure.git
+cd xmtpd-infrastructure/helm
+```
+
+Create a `xmtpd.yaml` file which sets all relevant `env` variables as described in the [Helm Readme](../helm/README.md)
+
+Additionally, you will need this section:
+```yaml
+# filename xmtpd.yaml
+# env: {}
+ingress:
+  create: true
+  host: <grpc.xmtp-partners.xyz>
+  tls:
+    certIssuer: letsencrypt-staging
+    secretName: xmtp-tls-cert
+```
+
+To install the helm chart, run
+```bash
+helm install xmtpd ./xmtpd -f xmtpd.yaml
+```
+
+Among other things, this will create an Ingress such as:
+```yaml
+# Source: xmtpd/templates/ingress.yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: xmtpd
+  labels:
+    helm.sh/chart: xmtpd-0.1.0
+    app.kubernetes.io/name: xmtpd
+    app.kubernetes.io/instance: xmtpd
+    app.kubernetes.io/version: "v0.1.1"
+    app.kubernetes.io/managed-by: Helm
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    cert-manager.io/cluster-issuer: letsencrypt-staging
+    nginx.ingress.kubernetes.io/ssl-redirect: "true"
+    nginx.ingress.kubernetes.io/backend-protocol: "GRPC"
+spec:
+  ingressClassName: nginx
+  rules:
+    - host: grpc.xmtp-partners.xyz
+      http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name:  xmtpd
+                port:
+                  number: 80
+  tls:
+  - secretName: xmtp-tls-cert
+    hosts:
+      - grpc.xmtp-partners.xyz
+```
+
+To validate, check for the existence of the Ingress:
+```bash
+kubectl get ing
+NAME               CLASS   HOSTS                          ADDRESS          PORTS     AGE
+xmtpd              nginx   grpc.xmtp-partners.xyz         35.202.107.198   80, 443   17h
+```
+
+It might take a few minutes for the Ingress to populate.
+
+In some cases neither the NGINX nor the cert-manager will create the secret.
+To work around this chicken-and-egg problem, you can create an empty secret such as:
```

Comment:
nitpick: should we simply create the resource as part of our helm chart?

Reply:
no. Because the reinstall will rewrite it and you get duplicate requests. It really screws `cert-manager` up... Best way to hit rate limits and lock yourself out for 3 days